### DISCUSSION TITLE
Do TemplateBindings support Mode=TwoWay or not?

### DISCUSSION THREAD

**Discussion by stogle** [AvaloniaUI/Avalonia#14887] [Q&A]
Hello,

According to the documentation at https://docs.avaloniaui.net/docs/guides/custom-controls/how-to-create-templated-controls#data-binding, "`TemplateBinding` only supports `OneWay` mode for performance reasons". However, in my testing it seems to work fine (at least in Avalonia 11.0.0), except that it does not seem to respect the property's default binding mode. I wonder if the documentation is out of date, or if there is a problem with `TwoWay` mode I'm not seeing.

My questions:
1. If I want to have a two-way binding in my control template, which is the "right" way to do it?
```
<ToggleButton IsChecked="{TemplateBinding IsChecked, Mode=TwoWay}" />
```
or
```
<ToggleButton IsChecked="{Binding IsChecked, RelativeSource={RelativeSource TemplatedParent}}" />
```
or
```
<ToggleButton IsChecked="{Binding $parent[MyControl].IsChecked}" />
```
2. Does the [documentation](https://docs.avaloniaui.net/docs/guides/custom-controls/how-to-create-templated-controls#data-binding) need to be updated?
3. Is it a bug that `TemplateBinding` does not respect default binding mode?

**Comment by maxkatz6** [comment]
Documentation needs to be updated, yes

**Reply by maxkatz6** [reply]
It supports OneWay and TwoWay only: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Data/TemplateBinding.cs#L156
